packages/@react-aria/tag/src/useTag.ts

```diff
@@ -73,34 +99,53 @@ export function useTag<T>(props: AriaTagProps<T>, state: ListState<T>, ref: RefO
   };
 
   let modality = useInteractionModality();
-  if (modality === 'virtual' &&  (typeof window !== 'undefined' && 'ontouchstart' in window)) {
-    modality = 'pointer';
+  if (
+    modality === "virtual" &&
+    typeof window !== "undefined" &&
+    "ontouchstart" in window
+  ) {
+    modality = "pointer";
   }
-  let description = onRemove && (modality === 'keyboard' || modality === 'virtual') ? stringFormatter.format('removeDescription') : '';
+  let description =
+    onRemove && (modality === "keyboard" || modality === "virtual")
+      ? stringFormatter.format("removeDescription")
+      : "";
   let descProps = useDescription(description);
 
   let isFocused = item.key === state.selectionManager.focusedKey;
+  let isNonFocused = state.selectionManager.focusedKey == null;
+  function getTabIndex(
+    isDisabled: boolean,
+    isFocused: boolean,
+    isNonFocused: boolean
+  ) {
+    if (isDisabled) {
+      return -1;
+    }
+    return isFocused || isNonFocused ? 0 : -1;
+  }
+
   let domProps = filterDOMProps(item.props);
   let linkProps = useSyntheticLinkProps(item.props);
   return {
     removeButtonProps: {
-      'aria-label': stringFormatter.format('removeButtonLabel'),
-      'aria-labelledby': `${buttonId} ${rowProps.id}`,
-      isDisabled: state.disabledKeys.has(item.key) || item.props.isDisabled,
+      "aria-label": stringFormatter.format("removeButtonLabel"),
+      "aria-labelledby": `${buttonId} ${rowProps.id}`,
+      isDisabled,
       id: buttonId,
-      onPress: () => onRemove ? onRemove(new Set([item.key])) : null,
-      excludeFromTabOrder: true
+      onPress: () => (onRemove ? onRemove(new Set([item.key])) : null),
+      excludeFromTabOrder: true,
     },
     rowProps: mergeProps(rowProps, domProps, linkProps, {
-      tabIndex: (isFocused || state.selectionManager.focusedKey == null) ? 0 : -1,
+      tabIndex: getTabIndex(isDisabled, isFocused, isNonFocused),
```

Comment:
Thank you for the opportunity to resolve this issue.
I solved an issue where the TabGroup component was deleted when it was disalbed.
I don't think it's a good code, but if there is a better code or something wrong, please point it out. thanks!